src/components/application_manager/src/app_service_manager.cc

```diff
@@ -81,19 +82,73 @@ smart_objects::SmartObject AppServiceManager::PublishAppService(
   service_record[strings::service_id] = service_id;
   service_record[strings::service_published] = true;
   service_record[strings::service_active] = true;
-
   app_service.record = service_record;
 
   published_services_.insert(
       std::pair<std::string, AppService>(service_id, app_service));
 
+  auto all_services = GetAllServices();
+  smart_objects::SmartObjectSPtr notification =
+      std::make_shared<smart_objects::SmartObject>(
+          MessageHelper::CreateMobileSystemCapabilityNotification(
+              all_services,
+              service_id,
+              mobile_apis::ServiceUpdateReason::PUBLISHED));
+  app_manager_.GetRPCService().ManageMobileCommand(
+      notification, commands::Command::CommandSource::SOURCE_SDL);
+  smart_objects::SmartObjectSPtr hmi_notification =
+      std::make_shared<smart_objects::SmartObject>(
+          MessageHelper::CreateHMISystemCapabilityNotification(
+              all_services,
+              service_id,
+              mobile_apis::ServiceUpdateReason::PUBLISHED));
+  app_manager_.GetRPCService().ManageHMICommand(hmi_notification);
   return service_record;
 }
 
 bool AppServiceManager::UnpublishAppService(const std::string service_id) {
   LOG4CXX_AUTO_TRACE(logger_);
-  // TODO
-  return false;
+
+  auto it = published_services_.find(service_id);
+  if (it == published_services_.end()) {
+    LOG4CXX_ERROR(logger_, "Service id does not exist in published services");
+    return false;
+  }
+  LOG4CXX_DEBUG(logger_, "Unpublishing app service: " << service_id);
+
+  SetServicePublished(service_id, false);
+
+  auto all_services = GetAllServices();
+  smart_objects::SmartObjectSPtr notification =
+      std::make_shared<smart_objects::SmartObject>(
+          MessageHelper::CreateMobileSystemCapabilityNotification(
+              all_services,
+              service_id,
+              mobile_apis::ServiceUpdateReason::REMOVED));
+  app_manager_.GetRPCService().ManageMobileCommand(
+      notification, commands::Command::CommandSource::SOURCE_SDL);
+  smart_objects::SmartObjectSPtr hmi_notification =
+      std::make_shared<smart_objects::SmartObject>(
+          MessageHelper::CreateHMISystemCapabilityNotification(
+              all_services,
+              service_id,
+              mobile_apis::ServiceUpdateReason::REMOVED));
+  app_manager_.GetRPCService().ManageHMICommand(hmi_notification);
+
+  published_services_.erase(it);
+
+  return true;
+}
+
+void AppServiceManager::UnpublishAppServices(const uint32_t connection_key) {
```

Comment:
I am uncertain as to whether this function would only be used by the `ApplicationManagerImpl::UnregisterApplication` function. At the moment we are only removing all the published services of an app when it unregisters but there might be other cases where we might have to do that. However to avoid confusion with `UnpublishAppService` I am changing it to `UnpublishServices`